app/models/concerns/variant_stock.rb

```diff
@@ -77,6 +77,18 @@ def on_demand=(new_value)
     end
   end
 
+  def on_demand_desired_or_current
+    return on_demand_desired if new_record?
+
+    on_demand
+  end
+
+  def on_hand_desired_or_current
+    return on_hand_desired if new_record?
+
+    on_hand
+  end
+
```

Comment:
These are arguably view-specific logic, so perhaps these could be moved to view helpers. Also `method_on_demand`.